hamster_gtk/overview/widgets/charts.py

```diff
@@ -34,65 +38,77 @@ class Charts(Gtk.Grid):
+        self.attach(Gtk.Label('Dates'), 2, 0, 1, 1)
+        self.attach(self._get_barcharts(totals.date), 2, 1, 1, 1)
 
-    def _get_category_widget(self, category_totals):
-        """Return a widget to represent all categories in a column."""
-        grid = Gtk.Grid()
-        row = 0
-        # the highest amount of time spend in a category. This is the scale for
-        # all other totals.
-        max_total = max(category_totals.values())
-
-        # Build individual category 'rows'.
-        for category, total in category_totals.items():
-            category_label = Gtk.Label(category)
-            bar_chart = HorizontalBarChart(total.total_seconds(), max_total.total_seconds())
-            total_label = Gtk.Label(self._get_delta_string(total))
-            grid.attach(category_label, 0, row, 1, 1)
-            grid.attach(bar_chart, 1, row, 1, 1)
-            grid.attach(total_label, 2, row, 1, 1)
-            row += 1
-        return grid
+        self.override_background_color(Gtk.StateType.NORMAL, Gdk.RGBA(1, 1, 1, 1))
 
-    # [FIXME]
-    # Place in a proper helper module. Maybe even in ``hamster-lib``?
-    # In that case make sure to check if we can refactor
-    # ``Fact.get_string_delta``!
-    def _get_delta_string(self, delta):
+    def _get_barcharts(self, totals):
         """
-        Return a human readable representation of ``datetime.timedelta`` instance.
+        Return a widget to represent all categories in a column.
 
-        In most contexts its not that useful to present the delta in seconds.
-        Instead we return the delta either in minutes or ``hours:minutes`` depending on the
-        value.
+        Args:
+            totals (dict): A dict that provides delta values for given keys. {key: delta}.
 
-        Note:
-            So far, this does not account for large deltas that span days and more.
+        Returns:
+            Gtk.Grid: A Grid which contains one column and as many rows as there
+                are ``keys`` in ``totals``. Each row contains a barchart with labels
+                showing the delta relative to the highest delta value in ``totals``.
         """
-        seconds = delta.total_seconds()
-        minutes = int(seconds / 60)
-        if minutes < 60:
-            result = '{} min'.format(minutes)
+        # The highest amount of time spend. This is the scale for all other totals.
+        # Python 2.7 does not yet have support for the ``default`` kwarg.
+        if not totals:
+            max_total = 0
         else:
-            result = '{hours:02d}:{minutes:02d}'.format(
-                hours=int(seconds / 3600), minutes=int((seconds % 3600) / 60))
-        return result
+            max_total = max(totals.values())
+            # Sorting a dict like this returns a list of tuples.
+            totals = sorted(totals.items(), key=operator.itemgetter(1),
+                            reverse=True)
+
+        grid = Gtk.Grid()
+        grid.set_column_spacing(5)
```

Comment:
Is it possible to use CSS for this? Maybe some selector like `#Overview grid>*`?